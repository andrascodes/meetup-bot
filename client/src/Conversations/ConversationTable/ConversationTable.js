import React from 'react'

import { ConversationRow } from './ConversationRow'

export const ConversationTable = props => {

  const rows = props.conversations.map(convo => (
    <ConversationRow 
      key={convo.id}
      handleRowClick={props.handleRowClick(convo.id)}
      {...convo}   
    />
  ))

  return (
    <table className="table table-hover">
      <thead>
        <tr>
          <th>participant</th>
          <th>date</th>
          <th>duration</th>
          <th>label</th>
          <th>errors</th>
        </tr>
      </thead>
      <tbody>
        {rows}
      </tbody>
    </table>
  )
}

ConversationTable.propTypes = {
  conversations: React.PropTypes.arrayOf(React.PropTypes.shape({
    id: React.PropTypes.number,
    participant: React.PropTypes.string,
    start: React.PropTypes.string,
    end: React.PropTypes.string,
    errors: React.PropTypes.string,
    label: React.PropTypes.string
  })).isRequired,
  handleRowClick: React.PropTypes.func.isRequired
}