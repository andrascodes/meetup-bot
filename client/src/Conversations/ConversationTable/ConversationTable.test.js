import React from 'react'
import renderer from 'react-test-renderer'

import { ConversationTable } from './ConversationTable'

describe('ConversationTable', () => {

  test('renders correctly', () => {
    
    const tree = renderer.create(
      <ConversationTable 
        handleRowClick={(id) => () => {}}
        conversations={[{
          id: 1,
          participant: '123',
          start: '2017-04-13 12:00',
          end: '2017-04-13 12:02',
          errors: 'OAuthException',
          label: 'label1 label2',
        }]}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  
  test('throws PropType error', () => {
    const render = () => renderer.create(
      <ConversationTable
        handleRowClick={(id) => () => {}}
      />
    )
    expect(render).toThrowError(`The prop 'conversations' has to be defined, pass an empty array ([])`)
  })

})