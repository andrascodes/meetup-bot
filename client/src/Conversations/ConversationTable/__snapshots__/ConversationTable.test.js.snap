exports[`ConversationTable renders correctly 1`] = `
<table
  className="table table-hover">
  <thead>
    <tr>
      <th>
        participant
      </th>
      <th>
        date
      </th>
      <th>
        duration
      </th>
      <th>
        label
      </th>
      <th>
        errors
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      id={1}
      onClick={[Function]}>
      <td>
        123
      </td>
      <td>
        April 13th 2017 12:00
      </td>
      <td>
        02:00
      </td>
      <td>
        label1 label2
      </td>
      <td>
        OAuthException
      </td>
    </tr>
  </tbody>
</table>
`;
